### PR TITLE
Add feature annotations to the reference page

### PR DESCRIPTION
**Pull request by berkayoz**
Add feature annotations to the reference page

### CODE REVIEW

File: docs/src/snap/reference/annotations.md [hunk 1]
```diff
@@ -8,6 +8,27 @@ the bootstrap configuration.
...
+| `k8sd/v1alpha1/calico/autodetection-v6/kubernetes` | Configure Calico to detect node addresses based on the Kubernetes API. | "NodeInternalIP" |
+| `k8sd/v1alpha1/calico/autodetection-v6/interface` | Enable IP auto-detection based on interfaces that match the given regex. | string |
+| `k8sd/v1alpha1/calico/autodetection-v6/skipInterface` | Enable IP auto-detection based on interfaces that do not match the given regex. | string |
+| `k8sd/v1alpha1/calico/autodetection-v6/canReach` | Enable IP auto-detection based on which source address on the node is used to reach the specified IP or domain. | string |
+| `k8sd/v1alpha1/calico/autodetection-v6/cidrs` | Enable IP auto-detection based on which addresses on the nodes are within one of the provided CIDRs. | []string (comma separated) |
+| `k8sd/v1alpha1/cilium/devices` | List of devices facing cluster/external network (used for BPF NodePort, BPF masquerading and host firewall); supports `+` as wildcard in device name, e.g. `eth+` | string |
```
Comment (louiseschmidtgen) [inline docs/src/snap/reference/annotations.md:26]:
```suggestion
| `k8sd/v1alpha1/cilium/devices` | List of devices facing cluster/external network (used for BPF NodePort, BPF masquerading and host firewall); supports `+` as wildcard in device name, e.g. `eth+,ens+` | string |
```
We could add a second string in the example to clarify the comma separation.

File: docs/src/snap/reference/annotations.md [hunk 2]
```diff
@@ -8,6 +8,27 @@ the bootstrap configuration.
...
+| `k8sd/v1alpha1/cilium/direct-routing-device` | Device name used to connect nodes in direct routing mode (used by BPF NodePort, BPF host routing; if empty, automatically set to a device with k8s InternalIP/ExternalIP or with a default route bridge type devices are ignored in automatic selection | string |
```
Comment (louiseschmidtgen) [inline docs/src/snap/reference/annotations.md:27]:
```suggestion
| `k8sd/v1alpha1/cilium/direct-routing-device` | Device name used to connect nodes in direct routing mode (used by BPF NodePort, BPF host routing. If the flag is undefined, device name is automatically set to a device with k8s InternalIP/ExternalIP or to a default route bridge. Type devices are ignored in automatic selection | string |
```
This doesn't flow quite right, maybe re-word it to something like this:

File: docs/src/snap/reference/annotations.md [hunk 3]
```diff
@@ -8,6 +8,27 @@ the bootstrap configuration.
...
+| `k8sd/v1alpha1/cilium/vlan-bpf-bypass` | Comma separated list of VLAN tags to bypass eBPF filtering on native devices. Cilium enables firewalling on native devices and filters all unknown traffic, including VLAN 802.1q packets, which pass through the main device with the associated tag (e.g., VLAN device eth0.4000 and its main interface eth0). Supports `0` as wildcard for bypassing all VLANs. | []string |
```
Comment (louiseschmidtgen) [inline docs/src/snap/reference/annotations.md:28]: Could you add an example for setting the flag please?

**Review by louiseschmidtgen — APPROVED** [review]
Thanks Berkay! Little nits in the comments

File: docs/src/snap/reference/annotations.md (hunk 2 above)
Comment (berkayoz) [reply docs/src/snap/reference/annotations.md]: This is from the upstream docs, seems like I messed up the formatting should be better now

**Comment by eaudetcobello** [comment]
I think you're missing k8sd/v1alpha1/csrsigning/auto-approve ?

**Review by eaudetcobello — APPROVED** [review]

File: docs/src/snap/reference/annotations.md [hunk 4]
```diff
@@ -8,6 +8,28 @@ the bootstrap configuration.
 |---------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------|
 | `k8sd/v1alpha/lifecycle/skip-cleanup-kubernetes-node-on-remove` | If set, only microcluster and file cleanup are performed.  This is helpful when an external controller (e.g., CAPI) manages the Kubernetes node lifecycle. By default,  k8sd will remove the Kubernetes node when it is removed from the cluster. | "true"\|"false" |
 | `k8sd/v1alpha/lifecycle/skip-stop-services-on-remove` | If set, the k8s services will not be stopped on the leaving node when removing the node. This is helpful when an external controller (e.g., CAPI) manages the Kubernetes node lifecycle. By default, all services are stopped on leaving nodes. | "true"\|"false" |
+| `k8sd/v1alpha1/csrsigning/auto-approve` | If set, certificate signing requests created by worker nodes are auto approved. | "true"\|"false" |
```
Comment (louiseschmidtgen) [inline docs/src/snap/reference/annotations.md:11]: Maybe @evilnick / @nhennigan can we change the width of the Description column it renders quite oddly:
image

**Review by louiseschmidtgen — COMMENTED** [review]
Last comment on formatting:

File: docs/src/snap/reference/annotations.md (hunk 4 above)
Comment (evilnick) [reply docs/src/snap/reference/annotations.md:11]: Will take a look but maybe a table isn't the best option here

**Pull request merged**